patches/server/0814-Overridable-YAML-configuration.patch

```diff
@@ -0,0 +1,330 @@
++     *                The group name must only contain a-z, A-Z, and 0-9. It may contain underscore (_),
++     *                however, it must not end with underscore. Underscore and dots will be appended to
++     *                the group name in the respective sources.
++     * @param sources array of {@link OverrideSource} to fetch overridden values from.
++     */
++    public OverridableYamlConfiguration(@NotNull String group, @NotNull OverrideSource[] sources) {
++        Preconditions.checkArgument(group.matches("[a-zA-Z0-9_]*[a-zA-Z0-9]+"));
++        Preconditions.checkNotNull(sources);
++        this.group = group;
++        this.sources = sources;
++    }
++
++    /**
++     * @param group the group name of this configuration. This will be used as a prefix in sources.
++     *              The group name must only contain a-z, A-Z, and 0-9. It may contain underscore (_),
++     *              however, it must not end with underscore. Underscore and dots will be appended to
++     *              the group name in the respective sources.
++     */
++    public OverridableYamlConfiguration(@NotNull String group) {
++        this(group, OverrideSource.SOURCES);
++    }
++
++    @Override
++    public @Nullable String getString(@NotNull String path, @Nullable String def) {
++        String override = getOverride(path);
++        if (override != null) {
++            return override;
++        }
++        return super.getString(path, def);
++    }
++
++    @Override
++    public int getInt(@NotNull String path, int def) {
++        String override = getOverride(path);
++        if (override != null) {
++            try {
++                return Integer.parseInt(override);
++            } catch (NumberFormatException ignored) {
++
++            }
++        }
++        return super.getInt(path, def);
++    }
++
++    @Override
++    public boolean getBoolean(@NotNull String path, boolean def) {
++        String override = getOverride(path);
++        if (override != null) {
++            return Boolean.parseBoolean(override);
++        }
++        return super.getBoolean(path, def);
++    }
++
++    @Override
++    public double getDouble(@NotNull String path, double def) {
++        String override = getOverride(path);
++        if (override != null) {
++            try {
++                return Double.parseDouble(override);
++            } catch (NumberFormatException ignored) {
++
++            }
++        }
++        return super.getDouble(path, def);
++    }
++
++    @Override
++    public long getLong(@NotNull String path, long def) {
++        String override = getOverride(path);
++        if (override != null) {
++            try {
++                return Long.parseLong(override);
++            } catch (NumberFormatException ignored) {
++
++            }
++        }
++        return super.getLong(path, def);
++    }
++
++    private @Nullable String getOverride(@NotNull String path) {
```

Comment:
I don't know if there is a better way to do this, but this method basically matches the logic found in the `static` version in `OverrideSource`. Idk if we could expand that method to also take the sources array (making it purely functional), allowing basically any implementation to just call the static method.